clients/src/main/java/org/apache/kafka/clients/consumer/internals/AbstractCoordinator.java

```diff
@@ -1102,8 +1102,13 @@ public GroupCoordinatorMetrics(Metrics metrics, String metricGrpPrefix) {
                 "The number of seconds since the last successful rebalance event"),
                 lastRebalance);
 
-            Measurable lastHeartbeat = (config, now) ->
-                TimeUnit.SECONDS.convert(now - heartbeat.lastHeartbeatSend(), TimeUnit.MILLISECONDS);
+            Measurable lastHeartbeat = (config, now) -> {
```

Comment:
This is a piggy-backed fix: if no heartbeat is triggered, we simply return `now` value which is meaningless as it is just since the epoch. So I've refactored to also return -1 as we did for the newly added last-rebalance.